### PR TITLE
jam-embedded article update

### PR DESCRIPTION
**Pull request by NickIliev**
https://kinvey.atlassian.net/browse/FJ-937

and 

https://kinvey.atlassian.net/browse/FJ-936

### CODE REVIEW

File: portal/jam-embedded-api.md [hunk 1]
```diff
@@ -92,231 +95,291 @@ The `share` method accepts an optional argument of the `ShareOptions` object typ
...
 
+Alternatively, you can use the `share` method without the optional `ShareOptions` argument. In that case, the log won't be encrypted and will be uploaded to the default organizational workspace.
 
-## Basic Implementations
 
-The following code snippets demonstrate a basic Fiddler Jam Embedded implementation that you can run through a test localhost server.
+## Basic Implementation
 
-If no file path is provided through the `serviceWorkerPath` argument, then the `service-worker.js` file must be on the same level as the `index.html` file below.
+The following code snippets demonstrate a basic Fiddler Jam Embedded implementation you can use as a reference and run through a test localhost server.
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:113]: This sounds like the implementation cannot be used for production server

File: portal/jam-embedded-api.md [hunk 2]
```diff
@@ -14,12 +14,6 @@ Fiddler Jam Embedded is a JavaScript library that provides an API to integrate t
 
 Follow the steps below to include and use the Fiddler Jam Embedded library in your webpage:
 
-1. Provide the list of endpoints that will integrate it (you can use Fiddler Jam Embedded only with a pre-set list of URLs)
-
-    ```CSV
-    https://getfiddler.com, https://docs.telerik.com
-    ```
-
 1. Obtain and store your unique Fiddler Jam Embedded API key.
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:17]: we need to explain how

File: portal/jam-embedded-api.md [hunk 3]
```diff
@@ -92,231 +95,291 @@ The `share` method accepts an optional argument of the `ShareOptions` object typ
...
-                </div>
-                <div>
-                    <input type="checkbox" id="captureStorage" name="captureStorage" />
-                    <label for="captureStorage">Capture Storage</label>
-                </div>
+                    <div>
+                        <input type="checkbox" id="captureDom" name="captureDom" />
+                        <label for="captureDom">Capture DOM</label><br />
+                        <label>With <b>captureDom = true</b>, and <b>captureVideo = true</b> (default values), you can start capturing with video recording (iframes are not recorded) without any user interaction (no explicit permission required).</label><br />
+                        <label>With <b>captureDom = false</b>, and <b>captureVideo = true</b>, you can start capturing with pixel-percet video recording (captures iframes), but also requires an explcit permission from the user.</label>
```
[the remark this answers is not in the document]
Comment (rosen-vladimirov) [reply portal/jam-embedded-api.md:167]: explcit -> explicit

File: portal/jam-embedded-api.md [hunk 4]
```diff
@@ -92,231 +95,291 @@ The `share` method accepts an optional argument of the `ShareOptions` object typ
...
+    // Generates and uploads a Fiddler Jam log share URL (async method). 
+    // The logs are automatically added to the default organizational workspace in the Fiddler Jam portal.
+    async function share() {
+        document.getElementById('capture-info').innerHTML = 'Processing generated log! (jam.state = ' + jam.state + ')';
+        document.getElementById('jam-share-url').innerHTML = 'Please wait! Uploading the log and generating share URL ...';
+
+        // The jam.share() method accepts argument of type { workspaceId: string, submittedBy: string, password: string }
+        const jamShareUrl = await jam.share(); // when the ShareOptions are omitted, the is uploaded to the default organization workspace without encryption protection.
+        document.getElementById('capture-info').innerHTML = 'Log generation completed! (jam.state = ' + jam.state + ')';
+        document.getElementById('jam-share-url').innerHTML = 'Share URL: ' +  '<a href="' + jamShareUrl + '">'+ jamShareUrl +'</a>';
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:337]: add ` target="_blank"`, so the link will not be opened in the same tab

File: portal/jam-embedded-api.md [hunk 5]
```diff
@@ -92,231 +95,291 @@ The `share` method accepts an optional argument of the `ShareOptions` object typ
...
-</html>
 
-```
 
 ## Limitations and Browser Specifics
 
 While incorporating your own Fiddler Jam Embedded tool into your website, note that there are some specifics related to different browsers and the core functionalities.
 
-- To capture a video recording without reloading a page, you can call the `start()` method (through `reloadPage: false` and `captureVideo: true`) on all browsers.
+1. You can **record a DOM video** while setting  `captureDom: true` and `captureVideo: true`. The DOM video recording is not pixel-perfect, and won't contain recordings from iframes. This method for video recording **does not require explicit confirmation** from the user.
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:367]: Isn't this the default method if you do not pass any configuration?

File: portal/jam-embedded-api.md [hunk 6]
```diff
@@ -14,46 +14,46 @@ Fiddler Jam Embedded is a JavaScript library that provides an API to integrate t
...
+
+    ```html
+        <html lang="en">
+            <head>
+                <meta charset='utf-8'>
+                <meta name='viewport' content='width=device-width,initial-scale=1'>
+            
+                <title>Fiddler Jam Embedded</title>
+
+                <script src="https://downloads.getfiddler.com/jam-embedded/fiddler-jam-embedded.js" id="jamEmbeddedScript" async></script>
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:31]: you can set crossorigin="anonymous" directly here inside the script tag.

File: portal/jam-embedded-api.md [hunk 7]
```diff
@@ -14,46 +14,46 @@ Fiddler Jam Embedded is a JavaScript library that provides an API to integrate t
...
+                <script>
+                    const jamEmbeddedScript = document.getElementById('jamEmbeddedScript');
+                    jamEmbeddedScript.crossOrigin = 'anonymous';  // set crossOrigin to enable CORS (and to be able to use the load event below)
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:34]: The correct name of the attribute is crossorigin: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin

File: portal/jam-embedded-api.md [hunk 8]
```diff
@@ -14,46 +14,46 @@ Fiddler Jam Embedded is a JavaScript library that provides an API to integrate t
...
 1. Create a `service-worker.js` file and import the service worker script from the following CDN link: https://downloads.getfiddler.com/jam-embedded/fje-service-worker.js.
 
-    By default, the `service-worker.js` file is expected to be on the same level as the HTML page that contains the `fiddler-jam-embedded.js` CDN script. You can change the default path of the worker file through the `InitOptions` object and its `serviceWorkerPath` property. Below you can find an example for importing the service worker from the Fiddler CDN.
+    By default, the `service-worker.js` file is expected to be on the same level as the HTML page that contains the `fiddler-jam-embedded.js` CDN script. You can change the default path of the worker file through the `InitOptions` object and its `serviceWorkerPath` property. Below is an example of importing the service worker from the Fiddler CDN.
     
-    ```JavaScript
-    self.importScripts(`https://downloads.getfiddler.com/jam-embedded/fje-service-worker.js`);
-    ```
+```JavaScript
+self.importScripts(`https://downloads.getfiddler.com/jam-embedded/fje-service-worker.js`);
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:55]: The format on the test docs site does not look good when ` are used, maybe you can use `"` here
Screen Shot 2022-07-14 at 9 09 51

File: portal/jam-embedded-api.md [hunk 9]
```diff
@@ -92,180 +92,83 @@ The `share` method accepts an optional argument of the `ShareOptions` object typ
...
+
+## Basic Implementation
+
+The following code snippets demonstrate a basic Fiddler Jam Embedded implementation.
+
+- Service worker file (for example, a file called `./service-worker.js`).
+
+    When no file path is provided through the `serviceWorkerPath` argument, then the `service-worker.js` file must be on the same level as the `index.html` file.
+
+Í
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:126]: strange symbol here

File: portal/jam-embedded-api.md [hunk 10]
```diff
@@ -92,180 +92,83 @@ The `share` method accepts an optional argument of the `ShareOptions` object typ
...
-
-                jam.addStateChangedEventListener(newState => {
-                    updateCaptureInfo();
-                });
-
-                initSettingsEvents();
+            const jamEmbeddedScript = document.getElementById('jamEmbeddedScript');
+            jamEmbeddedScript.crossOrigin = 'anonymous'; // set crossOrigin to enable CORS (and to be able to use the load event below)
+            jamEmbeddedScript.addEventListener('load', () => {
+                // Adding the custom implmenetaion
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:147]: implmenetaion -> implementation

File: portal/jam-embedded-api.md [hunk 11]
```diff
@@ -92,180 +92,83 @@ The `share` method accepts an optional argument of the `ShareOptions` object typ
...
 <html lang="en">
     <head>
         <meta charset='utf-8'>
         <meta name='viewport' content='width=device-width,initial-scale=1'>
     
         <title>Fiddler Jam Embedded</title>
 
-        <script src="https://downloads.getfiddler.com/jam-embedded/fiddler-jam-embedded.js"></script>
+        <!-- Loading the main Fiddler Jam Embedded source file asynchronously -->
+        <script src="https://downloads.getfiddler.com/jam-embedded/fiddler-jam-embedded.js" id="jamEmbeddedScript" async></script>
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:142]: set crossorigin="anonymous" directly here

File: portal/jam-embedded-api.md (hunk 10 above)
Comment (rosen-vladimirov) [reply portal/jam-embedded-api.md:147]: Also, we can add a comment that they can place the Jam Embedded implementation here as well, but we recommend them to use a separate file for easier maintenance

File: portal/jam-embedded-api.md [hunk 12]
```diff
@@ -278,45 +181,229 @@ The snippet uses a sample `index.html` page. Note that you will have to replace
...
+    await jam.start(captureOptions);
+
+    document.getElementById('btn-start').disabled = true;
+    document.getElementById('btn-stop').disabled = false;
+    document.getElementById('btn-share').disabled = false;
+    document.getElementById('capture-info').innerHTML = 'Started successfully!<br/>jam.state = ' + jam.state + ';<br/>captureVideo = ' + captureOptions.captureVideo +';<br/>videoFormat = "'+ captureOptions.videoFormat +'";<br/>captureScreenshots = ' + captureOptions.captureScreenshots +';<br/>captureConsole = ' + captureOptions.captureConsole +';<br/>captureStorage = ' + captureOptions.captureStorage +';';
+}
+
+// Starts pixel-perfect video recording (async method). 
+// Use this method when explicit permissoin is required to start video recording (for example, the uswer denied permission after calling start())
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:314]: the uswer -> the user

**Review by rosen-vladimirov — CHANGES_REQUESTED** [review]
I do not see any information about the video masking settings

File: portal/jam-embedded-api.md [hunk 13]
```diff
@@ -278,45 +240,253 @@ The snippet uses a sample `index.html` page. Note that you will have to replace
...
+
+- Navigate to the Fiddler Jam Portal menu (top-right corner) and follow the **Settings > Embedded** (the option is available only after the organizational quota is set).
+
+- Click on **API Key: Generate** link.
+
+- Copy your Fiddler Jam Embedded API key.
+
+If you want to regenerate the key, please get in touch with us at [support@getfiddler.com](mailto:support@getfiddler.com).
+
+## Limitations and Log Specifics
```
Comment (rosen-vladimirov) [inline portal/jam-embedded-api.md:466]: Can you please add information that masking is not supported for screenshots and also that there's a performance issue when capturing screenshots for large applications.

File: .gitignore [hunk 14]
```diff
@@ -23,7 +24,7 @@ Dockerfile
 Gemfile
 Gemfile.lock
 _site
-_tempconfig.yml
+_tempconfig.\`yml
```
Comment (rosen-vladimirov) [inline .gitignore:27]: Is this change intentional?

**Review by rosen-vladimirov — APPROVED** [review]

File: .gitignore (hunk 14 above)
Comment (NickIliev) [reply .gitignore:27]: actually no :) good catch!

**Pull request merged**